dropshot/src/error.rs

```diff
@@ -284,3 +286,15 @@ impl HttpError {
             .unwrap()
     }
 }
+
+impl fmt::Display for HttpError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "HttpError({}): {}", self.status_code, self.external_message)
+    }
+}
+
+impl Error for HttpError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
```

Comment:
This might be fine for what you're doing now, but it seems like there are many cases where we could provide a more useful source error.  In oxide-api-prototype, I would expect a lot of errors could have a `source` that was an `ApiError`.

Reply:
While that might logically be true, it doesn't currently seem possible in the context of this object's members.

From what I can tell, [source](https://doc.rust-lang.org/std/error/trait.Error.html#method.source) generally refers to "wrapping an sub-error" when crossing abstraction barriers - since `HttpError` doesn't contain an `ApiError`, nor does it have any way of accessing an `ApiError`, I think this would require modifying the contents (and construction) of `HttpErrors` to preserve the original error type.

Reply:
Yeah, I think that's right.  We could always add this later.